src/main/java/com/battleship/backend/models/Game.java

```diff
@@ -0,0 +1,105 @@
+package com.battleship.backend.models;
+
+import com.battleship.backend.BoardRepository;
+import com.battleship.backend.exceptions.GameNotReadyToStartException;
+import com.battleship.backend.validators.Validator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Component
+public class Game {
+    BoardRepository boardRepository;
+    ComputerPlayer computerPlayer;
+   final ArrayList<Ship> shipList;
+    Validator placeShipsValidator;
+    Validator hitRequestValidator;
+    boolean isGameStarted;
+    ArrayList<Ship> playerShipList;
+
+    public Game(BoardRepository boardRepository, Validator[] validators, Ship[] shipList){
+        this.boardRepository = boardRepository;
+        this.shipList = new ArrayList<Ship>(Arrays.stream(shipList).toList());
+        placeShipsValidator = validators[0];
+        hitRequestValidator = validators[1];
+        computerPlayer = new ComputerPlayer(boardRepository, shipList, placeShipsValidator, hitRequestValidator);
+        isGameStarted = false;
+        playerShipList = new ArrayList<Ship>(Arrays.stream(shipList).toList());
+    }
+
+    public void start(){
+        if(!isGameStarted){
+            setUpComputerBoard();
+            isGameStarted = true;
+        }
+    }
+
+    public void placeShip(Boardable board, Request placeRequest){
+        board.addShip(placeRequest.getShip(), placeRequest.getRow(), placeRequest.getCol());
+    }
+
+    public void fire(Boardable board, Request hitRequest){
+        board.hitSection(hitRequest.getRow(), hitRequest.getCol());
+    }
+
+    public boolean isPlayerReadyToStart(){
+        return getPlayerBoard().getShipList().equals(shipList) && playerShipList.isEmpty();
+    }
+
+    public void removePlayerShip(Ship ship){
+        playerShipList.remove(ship);
+    }
+
+    public void takeComputerTurn(){
+        Request hitRequest = computerPlayer.generateValidHitRequest();
+        fire(getPlayerBoard(), hitRequest);
+    }
+
+    private void setUpComputerBoard() {
+        Boardable computerBoard = boardRepository.getComputerBoard();
+        for (Ship ship : computerPlayer.getShipList()) {
+            Request placeRequest = computerPlayer.generateValidPlaceRequest(ship);
+            placeShip(computerBoard, placeRequest);
+            computerPlayer.removeShip(ship);
+        }
```

Comment:
Why did you choose to set up the computer board in the game vs the ComputerBoard?

Reply:
You mean have the board set itself up? Mostly because the board right now doesn't have access to the Computer Player, the factory, or the shipList.  I can refactor it so it works though.

Reply:
Just wondering.

Reply:
I decided not to move it, I think it makes more sense for the Game to setup the board.